core/src/batch/rxtx.rs

```diff
@@ -20,15 +22,18 @@ impl PacketTx for PortQueue {
     }
 }
 
-impl PacketRx for Vec<Mbuf> {
+impl PacketRx for Receiver<Mbuf> {
     fn receive(&mut self) -> Vec<Mbuf> {
-        self.drain(..).collect()
+        iter::from_fn(|| self.try_recv().ok()).collect::<Vec<_>>()
     }
 }
 
-impl PacketTx for Vec<Mbuf> {
+impl PacketTx for Sender<Mbuf> {
     fn transmit(&mut self, packets: Vec<Mbuf>) {
-        self.extend_from_slice(&packets)
+        packets.into_iter().for_each(|packet| {
+            let _ = self.send(packet);
+            ()
```

Comment:
think you don't need the `()` void when you have the let underscore before? playground seems to not warn.

Reply:
yep fixed the lint in the `emit` pr. forgot to push and carry over.